src/api/entities/Instruction/index.ts

```diff
@@ -382,6 +385,44 @@ export class Instruction extends Entity<UniqueIdentifiers, string> {
     });
   }
 
+  /**
+   * Retrieve current status of this Instruction
+   *
+   * @note uses the middlewareV2
+   */
+  public async getStatusV2(): Promise<InstructionStatusResult> {
+    const isPending = await this.isPending();
+
+    if (isPending) {
+      return {
+        status: InstructionStatus.Pending,
+      };
+    }
+
+    let eventIdentifier = await this.getInstructionEventFromMiddlewareV2(
+      EventIdEnum.InstructionExecuted
+    );
+    if (eventIdentifier) {
+      return {
+        status: InstructionStatus.Executed,
+        eventIdentifier,
+      };
+    }
+
+    eventIdentifier = await this.getInstructionEventFromMiddlewareV2(EventIdEnum.InstructionFailed);
```

Comment:
should this be done in parallel with the `InstructionExecuted` case?

With GraphQL it can be one query, but a `Promise.all` should be OK as well.